tests/windows_test.go

```diff
@@ -0,0 +1,235 @@
+	"kubevirt.io/kubevirt/pkg/kubecli"
+	"kubevirt.io/kubevirt/tests"
+)
+
+const (
+	windowsDisk       = "windows-disk"
+	windowsFirmware   = "5d307ca9-b3ef-428c-8861-06e72d69f223"
+	windowsVmUser     = "Administrator"
+	windowsVmPassword = "Heslo123"
+)
+
+const (
+	winrmCli    = "winrmcli"
+	winrmCliCmd = "winrm-cli"
+)
+
+var _ = Describe("Windows VM", func() {
+	flag.Parse()
+
+	virtClient, err := kubecli.GetKubevirtClient()
+	tests.PanicOnError(err)
+
+	var windowsVm *v1.VirtualMachine
+
+	gracePeriod := int64(0)
+	spinlocks := uint32(8191)
+	firmware := types.UID(windowsFirmware)
+	_false := false
+	windowsVmSpec := v1.VirtualMachineSpec{
+		TerminationGracePeriodSeconds: &gracePeriod,
+		Domain: v1.DomainSpec{
+			CPU: &v1.CPU{Cores: 2},
+			Features: &v1.Features{
+				ACPI: v1.FeatureState{},
+				APIC: &v1.FeatureAPIC{},
+				Hyperv: &v1.FeatureHyperv{
+					Relaxed:   &v1.FeatureState{},
+					VAPIC:     &v1.FeatureState{},
+					Spinlocks: &v1.FeatureSpinlocks{Retries: &spinlocks},
+				},
+			},
+			Clock: &v1.Clock{
+				ClockOffset: v1.ClockOffset{UTC: &v1.ClockOffsetUTC{}},
+				Timer: &v1.Timer{
+					HPET:   &v1.HPETTimer{Enabled: &_false},
+					PIT:    &v1.PITTimer{TickPolicy: v1.PITTickPolicyDelay},
+					RTC:    &v1.RTCTimer{TickPolicy: v1.RTCTickPolicyCatchup},
+					Hyperv: &v1.HypervTimer{},
+				},
+			},
+			Firmware: &v1.Firmware{UUID: firmware},
+			Resources: v1.ResourceRequirements{
+				Requests: k8sv1.ResourceList{
+					k8sv1.ResourceMemory: resource.MustParse("2048Mi"),
+				},
+			},
+			Devices: v1.Devices{
+				Disks: []v1.Disk{
+					{
+						Name:       windowsDisk,
+						VolumeName: windowsDisk,
+						DiskDevice: v1.DiskDevice{Disk: &v1.DiskTarget{Bus: "sata"}},
+					},
+				},
+			},
+		},
+		Volumes: []v1.Volume{
+			{
+				Name: windowsDisk,
+				VolumeSource: v1.VolumeSource{
+					Ephemeral: &v1.EphemeralVolumeSource{
+						PersistentVolumeClaim: &k8sv1.PersistentVolumeClaimVolumeSource{
+							ClaimName: tests.DiskWindows,
+						},
+					},
+				},
+			},
+		},
+	}
+
+	tests.BeforeAll(func() {
+		tests.SkipIfNoWindowsImage(virtClient)
+	})
+
+	BeforeEach(func() {
+		tests.BeforeTestCleanup()
+		windowsVm = tests.NewRandomVM()
+		windowsVm.Spec = windowsVmSpec
+	})
+
+	It("should success to start", func() {
+		vm, err := virtClient.VM(tests.NamespaceTestDefault).Create(windowsVm)
+		Expect(err).To(BeNil())
+		tests.WaitForSuccessfulVMStartWithTimeout(vm, 180)
+	}, 300)
+
+	It("should success to stop", func() {
+		By("Starting a VM")
+		vm, err := virtClient.VM(tests.NamespaceTestDefault).Create(windowsVm)
+		Expect(err).To(BeNil())
+		tests.WaitForSuccessfulVMStartWithTimeout(vm, 180)
+
+		By("Stopping the VM")
+		err = virtClient.VM(tests.NamespaceTestDefault).Delete(vm.Name, &metav1.DeleteOptions{})
+		Expect(err).To(BeNil())
+	}, 300)
+
+	Context("with winrm connection", func() {
+		var winrmcliPod *k8sv1.Pod
+		var cli []string
+		var output string
+
+		BeforeEach(func() {
```

Comment:
Ok I am convinced now that the BeforeAll which we have should also be good enough in this case.